### PR TITLE
[TFIDF] Remove duplicate arg, add cosine similarity utility function

### PR DESCRIPTION
**Pull request by spencerp**
**Patch description**
I found cosine similarity useful in a project, so I thought I'd add it here!

**Testing steps**
Set pdb trace and tested in interactive:
```
>>> q = tfidf_ranker.text2spvec(label)
>>> c = tfidf_ranker.text2spvec(sentence)
>>> cosine_similarity(q, c)
0.5348931739406508
```

### CODE REVIEW

File: parlai/agents/tfidf_retriever/tfidf_retriever.py [hunk 1]
```diff
@@ -81,12 +81,6 @@ def add_cmdline_args(
             type=int,
             help='How many docs to retrieve.',
         )
-        parser.add_argument(
-            '--retriever-mode',
```
Comment (klshuster) [inline parlai/agents/tfidf_retriever/tfidf_retriever.py:85]: why are we deleting this key?

**Review by klshuster — APPROVED** [review]
cool, lgtm

**Review by klshuster — CHANGES_REQUESTED** [review]
actually want to clarify the key removal prior to merge

File: parlai/agents/tfidf_retriever/tfidf_retriever.py (hunk 1 above)
Comment (spencerp) [reply parlai/agents/tfidf_retriever/tfidf_retriever.py:85]: It's a repeat of the arg on line 96 (90 in the new file)

**Review by klshuster — APPROVED** [review]

File: parlai/agents/tfidf_retriever/tfidf_retriever.py (hunk 1 above)
Comment (klshuster) [reply parlai/agents/tfidf_retriever/tfidf_retriever.py:85]: good catch!

**Comment by spencerp** [comment]
Checked `unittests_gpu16` and it doesn't appear to be related to these changes, so ignoring for the scope of this PR.

**Pull request merged**